apisix/init.lua

```diff
@@ -630,7 +630,7 @@ end
 function _M.http_header_filter_phase()
     core.response.set_header("Server", ver_header)
 
-    local status_code = tonumber(get_var("status"))
+    local status_code = tonumber(get_var("upstream_status"))
     if status_code then
```

Comment:
What about `if status_code and status_code >= 500 and status_code <= 599`

Reply:
nice, done.